packages/google_maps_flutter/lib/src/polyline.dart

```diff
@@ -0,0 +1,216 @@
+part of google_maps_flutter;
+
+/// Uniquely identifies a [Polyline] among [GoogleMap] polylines.
+///
+/// This does not have to be globally unique, only unique among the list.
+@immutable
+class PolylineId {
+  PolylineId(this.value) : assert(value != null);
+
+  /// value of the [PolylineId].
+  final String value;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+    if (other.runtimeType != runtimeType) return false;
+    final PolylineId typedOther = other;
+    return value == typedOther.value;
+  }
+
+  @override
+  int get hashCode => value.hashCode;
+
+  @override
+  String toString() {
+    return 'PolylineId{value: $value}';
+  }
+}
+
+/// Configuration options for [Polyline] instances.
+///
+/// When used to change configuration, null values will be interpreted as
+/// "do not change this configuration option".
+class Polyline {
+  const Polyline({
+    @required this.polylineId,
+    this.consumeTapEvents = false,
+    this.color,
+    this.endCap,
+    this.geodesic = false,
+    this.jointType,
+    this.points = const <LatLng>[],
+    this.pattern,
+    this.startCap,
+    this.visible = true,
+    this.width = 1,
+    this.zIndex = 0,
+    this.onTap,
+  });
+
+  /// Uniquely identifies a [Polyline].
+  final PolylineId polylineId;
+
+  /// If you want to handle events fired when the user taps the polyline, set this property to true.
```

Comment:
Please fix the comments in the file per the suggestion regarding effective dart documentation reference above.